lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -133,18 +133,30 @@ class CFunctionSignatureTypePrinter
     visitPart(aliasTy->getSinglyDesugaredType(), optionalKind, isInOutParam);
   }
 
+  void visitEnumType(EnumType *ET, Optional<OptionalTypeKind> optionalKind,
+                     bool isInOutParam) {
+    visitValueType(ET, optionalKind, isInOutParam);
+  }
+
   void visitStructType(StructType *ST, Optional<OptionalTypeKind> optionalKind,
                        bool isInOutParam) {
-    const StructDecl *SD = ST->getStructOrBoundGenericStruct();
+    visitValueType(ST, optionalKind, isInOutParam);
+  }
+
+  void visitValueType(NominalType *NT, Optional<OptionalTypeKind> optionalKind,
+                      bool isInOutParam) {
+    assert(isa<StructType>(NT) || isa<EnumType>(NT));
+    const auto *decl = NT->getNominalOrBoundGenericNominal();
+    assert(isa<StructDecl>(decl) || isa<EnumDecl>(decl));
 
     // Handle known type names.
-    if (printIfKnownSimpleType(SD, optionalKind, isInOutParam))
+    if (printIfKnownSimpleType(decl, optionalKind, isInOutParam))
       return;
     // FIXME: Handle optional structures.
```

Comment:
If I remember correctly, Swift optionals are enums. If so, can we say that optional structs are handled with this initial enum support, or do optionals still need some extra effort?

Reply:
Maybe, I’m not sure at the moment :) let’s keep the fixme for now until we have specific optional test coverage